doc/design/high-level.md

```diff
@@ -0,0 +1,220 @@
+# High Level Design
+
+rules_xcodeproj has a few high level design goals:
+
+- Only the `xcodeproj` rule is necessary to create a working project
+- The project can be further customized by using additional rules_xcodeproj
+  rules, or by returning certain rules_xcodeproj providers from custom rules
+- Outputs are as similar to `bazel build` as possible
+- The project feels as "native" in Xcode as possible
+- All definition of a project exists in `BUILD` files
+- The project can be configured to use either Xcode or Bazel as the build system
+
+It's also worth mentioning a few non-goals of rules_xcodeproj:
+
+- Providing additional build rules
```

Comment:
What exactly is a non-goal? Is something we strongly feel against or just a secondary goal? Because we then mention that the project can be customized with additional rules.

Reply:
That we are strongly against/won't do. This non-goal is talking about _build_ rules, which would go in your build graph, as opposed to the additional code generation rules.

I'll try to make that more clear.

Reply:
Nice, I got that after reading a bit more of the doc towards the end, but I think this change improves the clarity here too!